python-checks/src/main/java/org/sonar/python/checks/hotspots/GraphQLIntrospectionCheck.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.hotspots;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.Argument;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.ExpressionList;
+import org.sonar.plugins.python.api.tree.ListLiteral;
+import org.sonar.plugins.python.api.tree.RegularArgument;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.plugins.python.api.tree.Tuple;
+import org.sonar.python.tree.TreeUtils;
+
+@Rule(key = "S6786")
+public class GraphQLIntrospectionCheck extends PythonSubscriptionCheck {
+
+  private static final Set<String> GRAPHQL_VIEWS_FQNS = Set.of(
+    "flask_graphql.GraphQLView.as_view",
+    "graphql_server.flask.GraphQLView.as_view");
+
+  private static final Set<String> SENSITIVE_VALIDATION_RULE_FQNS = Set.of(
+    "graphene.validation.DisableIntrospection",
+    "graphql.validation.NoSchemaIntrospectionCustomRule");
+
+  private static final String MESSAGE = "Disable introspection on this GraphQL server endpoint.";
```

Comment:
Maybe change this to: 
`private static final String MESSAGE = "Disable introspection on this \"GraphQL\" server endpoint."; `

Reply:
This was the message in the RSPEC so not sure if I should change it...

Reply:
I think it's about the fact that we want to consistently highlight code in issue messages. That said, I can't (again) remember where this is supposed to be written, nor whether it should be backticks or double quotes...

Reply:
I would think here that GraphQL is just a brand. I am not sure I would consider it code. I.e. if it was Disable introspection on this Spring server endpoint. not sure I would highlight it. I can still make the change. The double quotes is the way to go.